application/comit_node/tests/rfc003_bitcoin_htlc.rs

```diff
@@ -3,21 +3,75 @@ use bitcoin_rpc_test_helpers::RegtestHelperClient;
 use bitcoin_support::{
     serialize::serialize_hex, Address, BitcoinQuantity, Network, OutPoint, PrivateKey, PubkeyHash,
 };
-use bitcoin_witness::{PrimedInput, PrimedTransaction};
-use comit_node::swap_protocols::rfc003::{bitcoin::Htlc, Secret};
+use bitcoin_witness::{
+    PrimedInput, PrimedTransaction, UnlockParameters, Witness, SEQUENCE_ALLOW_NTIMELOCK_NO_RBF,
+};
+use comit_node::swap_protocols::rfc003::{bitcoin::Htlc, Secret, SecretHash};
+use crypto::{digest::Digest, sha2::Sha256};
+use hex::FromHexError;
 use secp256k1_support::KeyPair;
+use spectral::prelude::*;
 use std::str::FromStr;
 use testcontainers::{clients::Cli, images::coblox_bitcoincore::BitcoinCore, Docker};
 
-fn fund_htlc(
+pub trait SecretTrait {
+    fn hash(&self) -> SecretHash;
+}
+
+impl SecretTrait for Secret {
+    fn hash(&self) -> SecretHash {
+        Secret::hash(self)
+    }
+}
+
+pub struct CustomSizeSecret(Vec<u8>);
+
+impl CustomSizeSecret {
+    pub fn unlock_with_secret(&self, htlc: &Htlc, keypair: KeyPair) -> UnlockParameters {
+        let public_key = keypair.public_key();
+        UnlockParameters {
+            witness: vec![
+                Witness::Signature(keypair),
+                Witness::PublicKey(public_key),
+                Witness::Data(self.0.to_vec()),
+                Witness::Bool(true),
+                Witness::PrevScript,
+            ],
+            sequence: SEQUENCE_ALLOW_NTIMELOCK_NO_RBF,
+            prev_script: htlc.script().clone(),
+        }
+    }
+}
+
+impl SecretTrait for CustomSizeSecret {
+    fn hash(&self) -> SecretHash {
+        let mut sha = Sha256::new();
+        sha.input(&self.0[..]);
+
+        let mut result: [u8; SecretHash::LENGTH] = [0; SecretHash::LENGTH];
+        sha.result(&mut result);
+        SecretHash::from(result)
+    }
+}
+
+impl FromStr for CustomSizeSecret {
+    type Err = FromHexError;
+
+    fn from_str(s: &str) -> Result<Self, <Self as FromStr>::Err> {
+        let secret = s.as_bytes().to_vec();
+        Ok(CustomSizeSecret(secret))
+    }
+}
+
+fn fund_htlc<S: SecretTrait>(
     client: &BitcoinCoreClient,
+    secret: &S,
```

Comment:
🔧  Just pass in the `SecretHash` and you can remove this trait. You could even remove `CustomSizeSecret` and just have a function which returns the unlock parameters for a given vec.

Reply:
I still need a `hash` function on `CustomSizeSecret` so I removed the trait but kept the struct.